pkg/store/consul/rcstore/consul_store.go

```diff
@@ -130,8 +131,11 @@ func (s *ConsulStore) Create(manifest manifest.Manifest, nodeSelector klabels.Se
 }
 
 // TODO: replace Create() with this
-func (s *ConsulStore) CreateTxn(txn *transaction.Tx, manifest manifest.Manifest, nodeSelector klabels.Selector, podLabels klabels.Set) (fields.RC, error) {
-	rc, err := s.innerCreateTxn(txn, manifest, nodeSelector, podLabels)
+func (s *ConsulStore) CreateTxn(ctx context.Context, manifest manifest.Manifest, nodeSelector klabels.Selector, podLabels klabels.Set) (fields.RC, error) {
+	rc, err := s.innerCreateTxn(ctx, manifest, nodeSelector, podLabels)
+	if err != nil {
```

Comment:
I suppose these three lines would hvae been good to have in the pre-context version of this code too